diesel/src/query_dsl/load_dsl.rs

```diff
@@ -91,6 +93,34 @@ where
     }
 }
 
+/// The `execute_with_returning_id` method
+///
+/// This is currently only implemented for [`MySQL`], because MySQL provides a
+/// `last_insert_id` field as part of its response packet to `INSERT` queries, but
+/// it's in a trait to avoid a backwards-incompatible change in the future if such
+/// functionality becomes available in other back-ends.
+///
+/// [`MySQL`]: crate::mysql::Mysql
+pub trait ExecuteInsertWithReturningId<
+    Conn: ConnectionWithReturningId<Backend = DB>,
+    DB: Backend = <Conn as Connection>::Backend,
+>: Sized
+{
+    /// Execute this command
```

Comment:
There should be a doc test/example on this function to show how this works

Reply:
Will do